core/src/main/java/org/apache/calcite/rel/core/RelFactories.java

```diff
@@ -303,6 +303,10 @@ public RelNode createAggregate(RelNode input,
   public interface FilterFactory {
     /** Creates a filter. */
     RelNode createFilter(RelNode input, RexNode condition);
+
+    /** Creates a filter. */
+    RelNode createFilter(RelNode input, RexNode condition,
```

Comment:
I'm not sure if this is "public API" or not, but if it is, this will be a "breaking change". Should it be documented somewhere? Another alternative could be providing a default implementation for the new method (which just ignores the CorrelationId set):
```
default RelNode createFilter(RelNode input, RexNode condition,
                             ImmutableSet<CorrelationId> variablesSet) {
  return createFilter(input, condition);
}
```
This approach will not break any existing implementation of `FilterFactory`, e.g. this patch would not need to modify `PigRelFactories.java`.
I'm not against the current solution, but I though it might be worth it to discuss this other alternative.

Reply:
Thanks for the suggestion. I think having a default method is a great idea since it wouldn't cause any compatibility issue and we don't need to provide additional implementations for other rels that don't have constructors for variablesSet. I updated the patch. Thanks

Reply:
I agree that a default method would be nice, but wouldn't that be misleading/causing runtime bugs in implementations as correlation id would not be populated anymore when using relfieldtrimmer, as the value is ignored?

Is it slightly better to check that the list is actually empty? (still a runtime exception but at least cause is clearly identified...)

Reply:
I've added comments in the JIRA case.